x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistry.java

```diff
@@ -405,7 +415,7 @@ private void addComposableTemplatesIfMissing(ClusterState state) {
                         currentTemplate.version(),
                         newTemplate.getValue().version()
                     );
-                    putComposableTemplate(templateName, newTemplate.getValue(), creationCheck);
+                    putComposableTemplate(state, templateName, newTemplate.getValue(), creationCheck, true);
```

Comment:
That's a very good point, but maybe since we are aiming to rely on https://github.com/elastic/elasticsearch/issues/96521 instead of this rollover mechanism in the long term, we can either decide to manually keep track of that, or simply decide we increase the version in the APM registry in every release until we have the full solution.
I think the bigger problem will be with the combination of the issue you raised below, where we need to know about dependencies versions as well...

@axw how is it with Fleet-managed index templates? Would Fleet know to rollover a data stream if its corresponding index template did not change, but it uses a stack-managed templates (like `logs-mappings`) that gets changed?

Reply:
@eyalkoren Fleet does not add any stack-managed templates to the index templates it installs at the moment. So it's not a problem there.

IMO we should defer solving this to #96521

Reply:
It doesn't do that at the moment but it soon will: https://github.com/elastic/kibana/issues/163141.

> Will there be a mechanism for rolling over a data stream if a component template changes?

IIRC a discussion with @axw and @eyalkoren we said that for now, we're fine with having to manually bump the version of the index template if a referenced component template changes.

https://github.com/elastic/elasticsearch/issues/96521 is the long-term solution.

Reply:
I edited my comment to explain this better. The APM registry uses the same version for all assets.
As @jbaiera implied, the problem is with templates that are coming from other registries